src/ws/cockpit-session-client-certificate.c

```diff
@@ -333,8 +308,23 @@ cockpit_session_client_certificate_map_user (const char *client_certificate_file
       return NULL;
     }
 
+  size_t my_cgroup_length;
+  char *my_cgroup = read_proc_self_cgroup (&my_cgroup_length);
+  if (my_cgroup == NULL)
+    {
+      warnx ("Could not determine cgroup of this process");
+      return NULL;
+    }
+  if (strncmp (cert_pem, my_cgroup, my_cgroup_length) != 0)
```

Comment:
Thinking aloud: This would also accept sub-cgroups of the expected one. However, (1) the additional suffix would then be part of `sssd_map_certificate()` and fail there (not even decodable as PEM), and (2) that should technically not even matter, as (hopefully) a cgroup child of the target ws instance can't get any more privileges than the parent. So LGTM.

Reply:
We include the newline we read from `/proc/self/cgroup` and include it in the comparison.  That could use some documentation.